File: modules/infra/src/main/java/hm/orz/chaos114/android/slideviewer/infra/model/TalkMetaData.java
package hm.orz.chaos114.android.slideviewer.infra.model;


import com.j256.ormlite.field.DatabaseField;

import lombok.Data;

@Data
public class TalkMetaData {
    @DatabaseField(generatedId = true)
    private Integer id;

    @DatabaseField(foreign = true, foreignAutoRefresh = true, columnName = "talk_id")
    private Talk talk;

    @DatabaseField
    private String title;

    @DatabaseField
    private String user;
}

File: modules/infra/src/main/java/hm/orz/chaos114/android/slideviewer/infra/model/TalkMetaData.kt
package hm.orz.chaos114.android.slideviewer.infra.model


import com.j256.ormlite.field.DatabaseField

data class TalkMetaData(
        @DatabaseField(generatedId = true)
        var id: Int? = null,

        @DatabaseField(foreign = true, foreignAutoRefresh = true, columnName = "talk_id")
        var talk: Talk? = null,

        @DatabaseField
        var title: String? = null,

        @DatabaseField
        var user: String? = null
)
